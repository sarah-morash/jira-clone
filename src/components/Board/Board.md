<!-- prettier- ignore -->
```
const sampleHeader = (
  <BoardHeader title="Sample Title" description="Sample Description" />
);

const sampleSwimlanes = [
  {
    id: "abc",
    title: "Swimlane 1",
    tickets: [
      {
        id: "abc",
        description: "Some",
        estimatedTime: 20,
        loggedTime: 10
      },
      {
        id: "bc",
        description: "Some",
        estimatedTime: 20,
        loggedTime: 10
      }
    ]
  },
  {
    id: "def",
    title: "Swimlane 2",
    tickets: [
      {
        id: "c",
        description: "Some",
        estimatedTime: 20,
        loggedTime: 10
      },
      {
        id: "d",
        description: "Some",
        estimatedTime: 20,
        loggedTime: 10
      }
    ]
  }
];

<Board header={sampleHeader} swimlanes={sampleSwimlanes} />
```
