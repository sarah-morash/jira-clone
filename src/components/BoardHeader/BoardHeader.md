<!-- prettier- ignore -->
```
<BoardHeader title="Title" description="some great description here" />
```
