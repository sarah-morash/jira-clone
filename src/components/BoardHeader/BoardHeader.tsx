import * as React from "react";
import styled from "react-emotion";

interface IBoardHeader {
  /**
   * The name of the board
   */
  title: string;

  /**
   * Description of the board
   */
  description: string;
}

const BoardHeaderWrapper = styled.div`
  padding-bottom: 20px;
`;

const BoardTitle = styled.h1`
  margin: 0;
`;

const BoardDescription = styled.p`
  margin: 0;
  padding-top: 5px;
`;

export default (props: IBoardHeader) => (
  <BoardHeaderWrapper>
    <BoardTitle> {props.title} </BoardTitle>
    <BoardDescription> {props.description} </BoardDescription>
  </BoardHeaderWrapper>
);
