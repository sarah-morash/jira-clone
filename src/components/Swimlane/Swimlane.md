<!-- prettier- ignore -->
```
const sampleTickets = [
  {
    id: "abc",
    description: "Some",
    estimatedTime: 20,
    loggedTime: 10
  },
  {
    id: "bc",
    description: "Some",
    estimatedTime: 20,
    loggedTime: 10
  }
];

<Swimlane title="Sample Title" tickets={sampleTickets} />
```
