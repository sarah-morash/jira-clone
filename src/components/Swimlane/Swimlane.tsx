import * as React from "react";
import styled from "react-emotion";
import Ticket, { ITicket } from "../Ticket/Ticket";

export interface ISwimlane {
  /**
   * Id
   */
  id: string;

  /**
   * Title
   */
  title: string;

  /**
   * Tickets
   */
  tickets: ITicket[];
}

const TicketWrapper = styled.div`
  &:nth-child(n + 2) {
    padding-top: 10px;
  }
`;

const SwimlaneWrapper = styled.div`
  background-color: #eeeeee;
  padding: 10px 5px;
`;

export default (props: ISwimlane) => (
  <SwimlaneWrapper>
    <div>{props.title.toUpperCase()}</div>
    {props.tickets.map((ticket: ITicket) => (
      <TicketWrapper key={ticket.id}>
        <Ticket {...ticket} />
      </TicketWrapper>
    ))}
  </SwimlaneWrapper>
);
