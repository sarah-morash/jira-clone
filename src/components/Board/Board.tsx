import * as React from "react";
import styled from "react-emotion";
import Swimlane, { ISwimlane } from "../Swimlane/Swimlane";

interface IBoard {
  /**
   * The header component
   * to be rendered atop
   * the board
   */
  header: JSX.Element;

  /**
   * List of Swimlane components
   * to be rendered
   */
  swimlanes: ISwimlane[];
}

const BoardWrapper = styled.div`
  padding: 20px;
  display: flex;
  min-height: 0;
  flex-direction: column;
  flex-grow: 1;
  flex-shrink: 0;
  flex-basis: 100%;
`;

const SwimlanesWrapper = styled.div`
  display: flex;
  flex-direction: row;
`;

interface ISwimlaneWrapper {
  /**
   * Number of swimlanes
   */
  numberLanes: number;
}

const SwimlaneWrapper = styled.div`
  flex: ${(props: ISwimlaneWrapper) =>
    `1 1 calc( 100% / ${props.numberLanes})`};

  &:nth-child(n + 2) {
    padding-left: 10px;
  }
`;

export default (props: IBoard) => (
  <BoardWrapper>
    {props.header}
    <SwimlanesWrapper>
      {props.swimlanes.map((swimlane: ISwimlane) => (
        <SwimlaneWrapper key={swimlane.id} numberLanes={props.swimlanes.length}>
          <Swimlane {...swimlane} />
        </SwimlaneWrapper>
      ))}
    </SwimlanesWrapper>
  </BoardWrapper>
);
