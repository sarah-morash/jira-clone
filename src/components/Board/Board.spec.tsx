import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import BoardHeader from "../BoardHeader/BoardHeader";
import Swimlane, { ISwimlane } from "../Swimlane/Swimlane";
import Board from "./Board";

describe("Board Component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const sampleHeader = (
      <BoardHeader title="Sample Title" description="Sample Description" />
    );
    const sampleSwimlanes: ISwimlane[] = [
      {
        id: "abc",
        title: "Swimlane 1",
        tickets: [
          {
            id: "abc",
            description: "Some",
            estimatedTime: 20,
            loggedTime: 10
          },
          {
            id: "bc",
            description: "Some",
            estimatedTime: 20,
            loggedTime: 10
          }
        ]
      },
      {
        id: "def",
        title: "Swimlane 2",
        tickets: [
          {
            id: "c",
            description: "Some",
            estimatedTime: 20,
            loggedTime: 10
          },
          {
            id: "d",
            description: "Some",
            estimatedTime: 20,
            loggedTime: 10
          }
        ]
      }
    ];

    wrapper = shallow(
      <Board header={sampleHeader} swimlanes={sampleSwimlanes} />
    );
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Header
   */
  it("Should display its header component atop the page", () => {
    expect(wrapper.find(BoardHeader)).toHaveLength(1);
  });

  /**
   * Swimlanes
   */
  it("Should create a <Swimlane /> component for each swimlane provided to its swimlanes prop", () => {
    expect(wrapper.find(Swimlane)).toHaveLength(2);
  });
});
