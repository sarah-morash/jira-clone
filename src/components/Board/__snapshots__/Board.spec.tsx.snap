// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component Should match its snapshot 1`] = `
<BoardWrapper>
  <Component
    description="Sample Description"
    title="Sample Title"
  />
  <SwimlanesWrapper>
    <SwimlaneWrapper
      key="abc"
      numberLanes={2}
    >
      <Component
        id="abc"
        tickets={
          Array [
            Object {
              "description": "Some",
              "estimatedTime": 20,
              "id": "abc",
              "loggedTime": 10,
            },
            Object {
              "description": "Some",
              "estimatedTime": 20,
              "id": "bc",
              "loggedTime": 10,
            },
          ]
        }
        title="Swimlane 1"
      />
    </SwimlaneWrapper>
    <SwimlaneWrapper
      key="def"
      numberLanes={2}
    >
      <Component
        id="def"
        tickets={
          Array [
            Object {
              "description": "Some",
              "estimatedTime": 20,
              "id": "c",
              "loggedTime": 10,
            },
            Object {
              "description": "Some",
              "estimatedTime": 20,
              "id": "d",
              "loggedTime": 10,
            },
          ]
        }
        title="Swimlane 2"
      />
    </SwimlaneWrapper>
  </SwimlanesWrapper>
</BoardWrapper>
`;
